govcd/edgegateway.go

```diff
@@ -145,15 +145,22 @@ func (eGW *EdgeGateway) AddDhcpPool(network *types.OrgVDCNetwork, dhcppool []int
 
 }
 
-// Temporary fix for existing Remove functions as it doesn't support network interfaces
-// TODO: remove this function when ruleId is available
-func (eGW *EdgeGateway) RemoveNATMappingRule(networkHref, natType, externalIP, internalIP, port string) (Task, error) {
-	return eGW.RemoveNATPortRule(networkHref, natType, externalIP, port, internalIP, port)
+// Deprecated in favor of RemoveNATRuleAsync, RemoveNATRule
+func (eGW *EdgeGateway) RemoveNATMapping(natType, externalIP, internalIP, port string) (Task, error) {
+	return eGW.RemoveNATPortMapping(natType, externalIP, port, internalIP, port)
 }
 
-// Temporary fix for existing Remove functions as it doesn't support network interfaces
-// TODO: remove this function when ruleId is available
-func (eGW *EdgeGateway) RemoveNATPortRule(networkHref, natType, externalIP, externalPort, internalIP, internalPort string) (Task, error) {
+// Deprecated in favor of RemoveNATRuleAsync, RemoveNATRule
```

Comment:
Same as above - would be nice to fix deprecation string.

Reply:
Fixed